frigate/detectors/plugins/rknn.py

```diff
@@ -1,118 +1,162 @@
+        self.rknn.release()
+
+    def get_soc(self):
         try:
             with open("/proc/device-tree/compatible") as file:
                 soc = file.read().split(",")[-1].strip("\x00")
         except FileNotFoundError:
-            logger.error("Make sure to run docker in privileged mode.")
             raise Exception("Make sure to run docker in privileged mode.")
 
         if soc not in supported_socs:
-            logger.error(
-                "Your SoC is not supported. Your SoC is: {}. Currently these SoCs are supported: {}.".format(
-                    soc, supported_socs
-                )
-            )
             raise Exception(
-                "Your SoC is not supported. Your SoC is: {}. Currently these SoCs are supported: {}.".format(
-                    soc, supported_socs
-                )
+                f"Your SoC is not supported. Your SoC is: {soc}. Currently these SoCs are supported: {supported_socs}."
             )
 
-        if not os.path.isfile("/usr/lib/librknnrt.so"):
-            if "rk356" in soc:
-                os.rename("/usr/lib/librknnrt_rk356x.so", "/usr/lib/librknnrt.so")
-            elif "rk3588" in soc:
-                os.rename("/usr/lib/librknnrt_rk3588.so", "/usr/lib/librknnrt.so")
-
-        self.core_mask = config.core_mask
-        self.height = config.model.height
-        self.width = config.model.width
-
-        if True:
-            os.makedirs("/config/model_cache/rknn", exist_ok=True)
-
-            if (config.model.width != 320) or (config.model.height != 320):
-                logger.error(
-                    "Make sure to set the model width and height to 320 in your config.yml."
+        return soc
+
+    def parse_model_input(self, model_path, soc):
+        model_props = {}
+
+        # find out if user provides his own model
+        # user provided models should be a path and contain a "/"
+        if "/" in model_path:
+            model_props["preset"] = False
+            model_props["path"] = model_path
+        else:
+            model_props["preset"] = True
+
+            """
+            Filenames follow this pattern:
+            origin-quant-basename-soc-tk_version-rev.rknn
+            origin: From where comes the model? default: upstream repo; rknn: modifications from airockchip
+            quant: i8 or fp16
+            basename: e.g. yolonas_s
+            soc: e.g. rk3588
+            tk_version: e.g. v2.0.0
+            rev: e.g. 1
+
+            Full name could be: default-fp16-yolonas_s-rk3588-v2.0.0-1.rknn
+            """
+
+            model_matched = False
+
+            for model_type, pattern in supported_models.items():
+                if re.match(pattern, model_path):
+                    model_matched = True
+                    model_props["model_type"] = model_type
+
+            if model_matched:
+                model_props["filename"] = model_path + f"-{soc}-v2.0.0-1.rknn"
+
+                # if model_path == default_model:
```

Comment:
can this be removed?

Reply:
It can be removed for now. However, I plan to include another model in the image that is freely usable. That's why I left the code there.

If it bothers you, I can remove it.

Reply:
personally I would prefer leave it out